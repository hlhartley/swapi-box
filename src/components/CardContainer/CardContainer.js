import React from 'react';
import '../../main.scss';
import Card from '../Card/Card';
// import { v4 } from 'uuid';

const CardContainer = (props) => {
    return (
        <div className="card-container">
            {props.category.map((property) => {
                return (<Card 
                    key={property.name} 
                    property={property} 
                    selected={props.selected} 
                    clickFavoriteButton={props.clickFavoriteButton}
                    favoritedCards={props.favoritedCards}
                />)
            })}
        </div>
    )
}

export default CardContainer;