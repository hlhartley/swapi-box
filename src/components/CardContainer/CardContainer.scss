.card-container {
    padding-top: 30px;
    display: flex;
    flex-flow: wrap;
    justify-content: space-evenly;
    color: black;
}
