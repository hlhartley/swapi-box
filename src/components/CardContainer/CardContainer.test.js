import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

let wrapper;
describe('CardContainer', () => {

  const cards = [{ 
    climate: "temperate", 
    created: "2014-12-10T11:35:48.479000Z",
    diameter: "12500",
    edited: "2014-12-20T20:58:18.420000Z",
    films: ["https://swapi.co/api/films/6/", "https://swapi.co/api/films/1/"],
    gravity: "1 standard",
    name: "Alderaan",
    orbital_period: "364",
    population: "2000000000",
    residents: ["https://swapi.co/api/people/5/", "https://swapi.co/api/people/68/", "https://swapi.co/api/people/81/"],
    rotation_period: "24",
    surface_water: "40",
    terrain: "grasslands, mountains",
    url: "https://swapi.co/api/planets/2/"
  }]

  beforeEach(() => {
    wrapper = shallow(<CardContainer
      category={cards} 
      selected={'people'} 
      clickFavoriteButton={() => {}}
    />); 
  })

    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.unmountComponentAtNode(div);
    })

    it('should exist', () => {   
      expect(wrapper).toBeDefined()
    })

    it('should match the snapshot with all data passed in correctly', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('should render the Card component', () => {
      expect(wrapper.find('Card').length).toEqual(1)
    })
  })
