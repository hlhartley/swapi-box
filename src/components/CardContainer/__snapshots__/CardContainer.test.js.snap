// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all data passed in correctly 1`] = `
<div
  className="card-container"
>
  <Card
    clickFavoriteButton={[Function]}
    key="Alderaan"
    property={
      Object {
        "climate": "temperate",
        "created": "2014-12-10T11:35:48.479000Z",
        "diameter": "12500",
        "edited": "2014-12-20T20:58:18.420000Z",
        "films": Array [
          "https://swapi.co/api/films/6/",
          "https://swapi.co/api/films/1/",
        ],
        "gravity": "1 standard",
        "name": "Alderaan",
        "orbital_period": "364",
        "population": "2000000000",
        "residents": Array [
          "https://swapi.co/api/people/5/",
          "https://swapi.co/api/people/68/",
          "https://swapi.co/api/people/81/",
        ],
        "rotation_period": "24",
        "surface_water": "40",
        "terrain": "grasslands, mountains",
        "url": "https://swapi.co/api/planets/2/",
      }
    }
    selected="people"
  />
</div>
`;
